app/controllers/UserController.scala

```diff
@@ -139,4 +141,19 @@ class UserController @Inject() (implicit val env: Environment[User, SessionAuthe
       }
     )
   }
+
+ // Post function that receives a JSON object with userId and isChecked, and updates the user's volunteer status.
+  def updateVolunteerStatus() = Action(parse.json) { request =>
+      val userId = (request.body \ "userId").as[UUID]
+      val isChecked = (request.body \ "isChecked").as[Boolean]
+
+      // Call the setCommunityService method from UserRoleTable
+      val result = UserRoleTable.setCommunityService(userId, isChecked)
```

Comment:
In Scala let's generally try to include data types and descriptive names! So I'd write the line as:
```
val rowsUpdated: Int = UserRoleTable.setCommunityService(userId, isChecked)
```